src/genie/libs/parser/iosxr/show_dampening.py

```diff
@@ -0,0 +1,241 @@
+            else:
+                continue
+            
+            m = p1.match(line)
+            
+            if m:
+                group = m.groupdict()
+                interface=group['interface'].rstrip()
+                
+                # set "not_present" on any empty fields
+                for empty in group.keys():
+                    if group.get(empty).startswith(' '):
+                        group[empty]="not_present"
+                protocol=group['prot'].rstrip()
+                
+                if result_dict['interface'].get(interface)==None:
+                    result_dict['interface'][interface]={}
+                    result_dict['interface'][interface]['protocol']={}
+                result_dict['interface'][interface]['protocol'][protocol]={}
+                result_dict['interface'][interface]['protocol'][protocol].update({'capsulation': group['cap']})  
+                result_dict['interface'][interface]['protocol'][protocol].update({'penalty': int(group['pen'])})
+                result_dict['interface'][interface]['protocol'][protocol].update({'suppressed': group['sup'].lower()})                                     
+                continue
+        return result_dict
+
+class ShowImDampeningIntfSchema(MetaParser):
+    schema = {
+        'interface': {
+            Any(): {
+            'dampening_status': str,    
+            Optional('currently_suppressed'): str,
+            Optional('half_life'): int,
+            Optional('max_supress_time'): int,
+            Optional('penalty'): int,
+            Optional('reuse'): int,
+            Optional('suppress'): int,
+            Optional('suppressed_secs_remaining'): int,
+            Optional('underlying_state'): str,
+            Optional('protocol'): 
+                     {Any():
+                        {Optional('protocol_capsulation'): str, 
+                        Optional('protocol_penalty'): int,
+                        Optional('protocol_suppression'): str,
+                        Optional('protocol_suppression_timer'): int,
+                        Optional('protocol_underlying_state'): str,
+                      },
+                   },
+                }
+        }
+    }
+    
+class ShowImDampeningIntf(ShowImDampeningIntfSchema):
+    """ Parser for show im dampening interface {interface} """
+    
+    cli_command = ['show im dampening interface {interface}']
+    
+    def cli(self, interface=None, output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            if interface:
+                out = self.device.execute(self.cli_command[1].format(
+                                          interface=interface))
+        else:
+            out = output
+        
+        result_dict = {}
+        result_dict.setdefault('interface', {})
```

Comment:
Same as the other parser. We want an empty dictionary unless the parser actually parses lines

Reply:
Yeah, I've moved that inside the match p1 now instead. Cheers.